packages/core/src/scd-plugin/stand-alone-mode/edition-actions.stand-alone-mode.ts

```diff
@@ -0,0 +1,82 @@
+// OPENSCD
+import {
+	isMove,
+	isCreate,
+	isDelete,
+	isReplace,
+	isUpdate
+} from '../../scd-events-v1'
+// UTILS
+import { serializeXmlDocument } from '../../scd-xml'
+// TYPES
+import type { EditorPluginInstance } from '../types.scd-plugin'
+import type { Create, Delete, Move, Replace, Update } from '../../scd-events-v1'
+
+//====== PRIVATE FUNCTIONS ======//
+
+function onMove(action: Move) {
+	console.warn('TODO onCreate', action)
+}
+
+function onCreate(pluginElement: EditorPluginInstance, action: Create) {
+	action.new.parent.insertBefore(
+		action.new.element,
+		action.new.reference ?? null
+	)
+	pluginElement.editCount++
+}
+
+function onDelete(action: Delete) {
+	console.warn('TODO onDelete', action)
+}
+
+function onReplace(action: Replace) {
+	console.warn('TODO onReplace', action)
+}
+
+function onUpdate(pluginElement: EditorPluginInstance, action: Update) {
+	for (const attribute of action.element.attributes) {
+		action.element.removeAttributeNode(attribute)
+	}
+	for (const [key, value] of Object.entries(action.newAttributes)) {
+		action.element.setAttribute(key, value ?? '')
+	}
+	pluginElement.editCount++
+}
+
+//====== PUBLIC FUNCTIONS ======//
+
+export function handleOpenDoc(
+	pluginElement: EditorPluginInstance,
+	event: CustomEvent
+) {
+	pluginElement.doc = event.detail.doc
+	pluginElement.docName = event.detail.docName
+}
+
+export function handleSaveDoc(pluginElement: EditorPluginInstance) {
```

Comment:
Its better to have a try/catch here, especially on url creation step

Reply:
We should have a common discussion on error handling, I will create a new US for this point.